### ISSUE TITLE
Wrong tag added if the same slug already exists

### ISSUE THREAD

**Issue by Simounet** [wallabag/wallabag#3302]
Hi there,
On wallabag.it, if I try to add the `cache` tag then validate it, it is switched to the `caché` tag word. I think the person creating a word with a slug that defining other words wins.
It could be great to have some kind of rules for special chars.

**Comment by Kdecherf** [comment]
This bug does not seem to be present on current master.

**Comment by j0k3r** [comment]
I guess you fixed it @Kdecherf here: https://github.com/wallabag/wallabag/pull/3139

**Comment by nicosomb** [comment]
@Simounet could you try master and tell us if the bug still appears?

**Comment by Kdecherf** [comment]
@nicosomb are you able to reproduce this issue on current version of wallabag.it?

**Comment by nicosomb** [comment]
Still on wallabag.it 😞

**Comment by Simounet** [comment]
Sorry for my late reply. I can confirm that I'm still encountering this issue on https://github.com/wallabag/wallabag/commit/1d91f14516e109978f497e9f55d2f87c8268e1d7.

**Comment by Kdecherf** [comment]
Can reproduce it on MySQL environment. PostgreSQL (_and SQLite for development_) seems to be unaffected.

**Comment by Kdecherf** [comment]
```
MariaDB [wallabag1]> SELECT * FROM wallabag_tag WHERE label = 'cache';
+----+--------+-------+
| id | label  | slug  |
+----+--------+-------+
|  1 | caché  | cache |
+----+--------+-------+
1 row in set (0.00 sec)
```

Well…

**Comment by Kdecherf** [comment]
Specifying a bin collate in the SELECT query seems to do the trick:

```
MariaDB [wallabag1]> SELECT * FROM wallabag_tag WHERE label = 'cache' collate utf8mb4_bin;
Empty set (0.00 sec)
```

**Comment by Kdecherf** [comment]
There are two fixes possible here:

- Implement the COLLATE word for `where` clause using a custom Doctrine query function and overload `TagRepository#findOneByLabel()` to call COLLATE() when using MySQL connector
- Convert `wallabag_tag.label` to collation `utf8_bin` instead of `utf8_unicode_ci`

The second fix is far easier than the first and more stable. But I don't know if there was a reason to use a case insensitive collation on this column in the first place.

@wallabag/core thoughts?

**Comment by j0k3r** [comment]
Second one should be more simpler.
And I don't know for the insensitive collation ...

**Comment by Kdecherf** [comment]
I don't know. But since we moved all tags to lowercase, I guess it's not useful anymore.

**Comment by nicosomb** [comment]
OK, so let's go for the 2nd fix.

**Comment by nicosomb** [comment]
Because we need a migration, we plan this fix for 2.4.0 (we only push migrations on minor releases).

**Comment by Strubbl** [comment]
I can reproduce the problem with tagging rules. I have a tag "notFound", which gets automatically assigned via tagging rule to an article if wallabag can't retrieve it. This worked in past, but since some time we have only lowercase tags. So everytime the tagging rule is triggered and applied to a new article a new tag gets created. The tag label is the same, it's "notfound", but i see the URL to delete is contains notfound-1 or notfound-2 etc. 

My workaround is to lowercase the tag name in the tagging rules.
